extensions/Worker.Extensions.CosmosDB/src/Config/CosmosDBBindingOptions.cs

```diff
@@ -48,8 +48,8 @@ internal virtual CosmosClient GetClient(string preferredLocations = "")
             // Do not override if preferred locations is configured via CosmosClientOptions
             if (!string.IsNullOrEmpty(preferredLocations) && CosmosExtensionOptions.ClientOptions.ApplicationPreferredRegions is null)
             {
-                // Copy to avoid modifying the original options
-                var cosmosClientOptions = CosmosExtensionOptions.ClientOptions;
+                // Clone to avoid modifying the original options
+                var cosmosClientOptions = CosmosExtensionOptions.Clone().ClientOptions;
```

Comment:
We don't need a clone of extension options, but rather client options:

``` CSharp
CosmosClientOptions cosmosClientOptions = (CosmosClientOptions) CosmosExtensionOptions.ClientOptions.MeberwiseClone();
cosmosClientOptions.ApplicationPreferredRegions = Utilities.ParsePreferredLocations(preferredLocations);
return ClientCache.GetOrAdd(cacheKey, (c) => CreateService(cosmosClientOptions));
```

Reply:
Ah okay. We will need to either: create a manual copy of `CosmosClientOptions`, or use reflection to invoke `MemberwiseClone`. Calling clone on the extension options will not work, as you will be getting the same instance of client options. This means we are still mutating the original instance when setting preferred locations, which is what we need to address.

Reply:
@jviau  I went with reflection via `CloningExtensions.cs` - let me know what you think